homeassistant/components/decora_wifi/config_flow.py

```diff
@@ -0,0 +1,107 @@
+"""Config flow for Decora Wifi integration."""
+
+import logging
+from typing import Any
+
+from decora_wifi import DecoraWiFiSession
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import DOMAIN as HOMEASSISTANT_DOMAIN, HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+BASE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class DecoreWifiConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for Decora Wifi Integration."""
+
+    VERSION = 1
+
+    async def async_step_import(self, import_data: dict[str, str]) -> FlowResult:
+        """Handle importting decora wifi config from configuration.yaml."""
+
+        async_create_issue(
+            self.hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            breaks_in_ha_version="2024.6.0",
+            is_fixable=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": "Decora Wifi",
+            },
+        )
```

Comment:
So we should be more specific here. According to ADR-21, we have to raise issues to tell the user what went wrong. Currently you will always tell the user YAML is deprecated and that its imported, even though it might have failed.

So we should raise separate issues on events like not being able to connect to decora or incorrect username/password.

When this happens, we should abort the flow, rather than return with errors (what the current code is doing since we return with errors)

Reply:
Check `nhc` for an example, or `nmbs`